tests/MongoDB.Driver.Tests/Specifications/Runner/MongoClientJsonDrivenTestRunnerBase.cs

```diff
@@ -299,19 +297,14 @@ protected virtual bool TryConfigureClientOption(MongoClientSettings settings, Bs
             return true;
         }
 
-        protected virtual void VerifyCollectionData(IEnumerable<BsonDocument> expectedDocuments)
-        {
-            VerifyCollectionData(expectedDocuments, null);
-        }
-
-        protected virtual void VerifyCollectionOutcome(BsonDocument outcome)
+        protected virtual void VerifyCollectionOutcome(BsonDocument outcome, string databaseName, string collectionName)
```

Comment:
The JSON test contains the name of the collection to verify. Refactor this method to:

        protected virtual void VerifyCollectionOutcome(BsonDocument outcome)
        {
            var collectionName = CollectionName;

            foreach (var aspect in outcome)
            {
                switch (aspect.Name)
                {
                    case "name":
                        collectionName = aspect.Value.AsString;
                        break;

                    case "data":
                        VerifyCollectionData(aspect.Value.AsBsonArray.Cast<BsonDocument>(), DatabaseName, collectionName);
                        break;

                    default:
                        throw new FormatException($"Unexpected collection outcome aspect: {aspect.Name}.");
                }
            }
        }